docs/architecture/adr-071-proposer-based-timestamps.md

```diff
@@ -250,84 +163,55 @@ type TimestampParams struct {
 A goal of adding proposer-based timestamps is to enforce some degree of clock synchronization, so having a mechanism that completely ignores the Unix time of the validator time no longer works.
 
 Validator clocks will not be perfectly in sync.
-Therefore, the proposer’s current known Unix time may be less than the `LastCommit.Timestamp`.
-If the proposer’s current known Unix time is less than the `LastCommit.Timestamp`, the proposer will sleep until its known Unix time exceeds `LastCommit.Timestamp`.
+Therefore, the proposer’s current known Unix time may be less than the previous block's `Header.Time`.
+If the proposer’s current known Unix time is less than the previous block's `Header.Time`, the proposer will sleep until its known Unix time exceeds it.
 
 This change will require amending the [defaultDecideProposal](https://github.com/tendermint/tendermint/blob/822893615564cb20b002dd5cf3b42b8d364cb7d9/internal/consensus/state.go#L1180) method.
-This method should now block until the proposer’s time is greater than `LastCommit.Timestamp`.
+This method should now block until the proposer’s time is greater than the previous block's `Header.Time`.
 
 #### Changes to the propose step timeout
 
 Currently, a validator waiting for a proposal will proceed past the propose step if the configured propose timeout is reached and no proposal is seen.
-Proposer-based timestamps requires changing this timeout logic. 
+Proposer-based timestamps requires changing this timeout logic.
 
-The proposer will now wait until its current known Unix time exceeds the `LastCommit.Timestamp` to propose a block.
+The proposer will now wait until its current known Unix time exceeds the previous block's `Header.Time` to propose a block.
 The validators must now take this and some other factors into account when deciding when to timeout the propose step.
 Specifically, the propose step timeout must also take into account potential inaccuracy in the validator’s clock and in the clock of the proposer.
 Additionally, there may be a delay communicating the proposal message from the proposer to the other validators. 
 
-Therefore, validators waiting for a proposal must wait until after the `LastCommit.Timestamp` before timing out.
-To account for possible inaccuracy in its own clock, inaccuracy in the proposer’s clock, and message delay, validators waiting for a proposal will wait until `LastCommit.Timesatmp + 2*ACCURACY + MSGDELAY`.
+Therefore, validators waiting for a proposal must wait until after the previous block's `Header.Time` before timing out.
+To account for possible inaccuracy in its own clock, inaccuracy in the proposer’s clock, and message delay, validators waiting for a proposal will wait until `Header.Time + 2*ACCURACY + MSGDELAY`.
```

Comment:
I don't think this is accurate. While is true the propose step should not timeout before the maximum receiving time for a `Proposal` to be considered `timely`, a `Proposal` can be received before `timeout_propose` and be `untimely`.

Reply:
Another point is that, when entering the `propose` round step, only the proposer knows `Header.Time`. Other validators must use their local time to schedule the propose timeout.

Reply:
> I don't think this is accurate. While is true the propose step should not timeout before the maximum receiving time for a Proposal to be considered timely, a Proposal can be received before timeout_propose and be untimely.

This section of the ADR is just discussing the propose phase timeout conditions. The timeout conditions for the propose phase are somewhat different from the conditions for agreeing that a block is timely. If you timeout _before_ seeing a block, you should prevote nil. If you did not timeout before receiving a block, you should then proceed to check the block for `timely`.

The conditions for checking timely are discussed [here](https://github.com/tendermint/tendermint/blob/62f3bccb655560d2b712cd429af4bb02977d97b7/docs/architecture/adr-071-proposer-based-timestamps.md#L221)

To clarify, the `Header.Time` that is meant in this line is the previous block's `Header.Time` which is known to the validators.